runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -250,7 +249,12 @@ State waitUntilFinish(
       }
 
       if (!hasError) {
+        // Reset the backoff, and update the new cumulative sleep time to be the remaining
+        // part of the total input sleep duration.
         backoff.reset();
+        long nanosConsumed = nanoClock.nanoTime() - startNanos;
+        Duration consumed = Duration.millis((nanosConsumed + 999999) / 1000000);
+        backoff = backoff.withMaxCumulativeBackoff(duration.minus(consumed));
```

Comment:
I agree BackOff shouldn't see the cumulative time.
But, here we are using FlexibleBackoff, and update MaxCumulativeBackoff each time.

It will also become easier to test if we put the tracking in FlexibleBackoff.